File: MSDKRemote/app/src/main/java/com/msdkremote/MyApplication.java
package com.msdkremote;

import android.app.Application;
import android.content.Context;
import android.util.Log;

import dji.v5.common.error.IDJIError;
import dji.v5.common.register.DJISDKInitEvent;
import dji.v5.manager.SDKManager;
import dji.v5.manager.interfaces.SDKManagerCallback;

public class MyApplication extends Application
{
    private final String TAG = this.getClass().getSimpleName();

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        // Before calling install, do not call any MSDK related
        // Basically, chines viruses
        com.secneo.sdk.Helper.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        // Initialize MSDK. It is recommended to put the initialization logic in Application.
        // Of course, you can put it anywhere according to your needs.
        SDKManager.getInstance().init(
            this,
            new SDKManagerCallback() {
                @Override
                public void onInitProcess(DJISDKInitEvent event, int totalProcess) {
                    Log.i(TAG, "onRegisterSuccess:");
                    if (event == DJISDKInitEvent.INITIALIZE_COMPLETE) {
                        SDKManager.getInstance().registerApp();
                    }
                }
                @Override
                public void onRegisterSuccess() {
                    Log.i(TAG, "onRegisterSuccess: ");
                }
                @Override
                public void onRegisterFailure(IDJIError error) {
                    Log.i(TAG, "onRegisterFailure: ");
                }
                @Override
                public void onProductConnect(int productId) {
                    Log.i(TAG, "onProductConnect: ");
                }
                @Override
                public void onProductDisconnect(int productId) {
                    Log.i(TAG, "onProductDisconnect: ");
                }
                @Override
                public void onProductChanged(int productId) {
                    Log.i(TAG, "onProductChanged: ");
                }
                @Override
                public void onDatabaseDownloadProgress(long current, long total) {
                    Log.i(TAG, String.format("onDatabaseDownloadProgress: %d / %d", current, total));
                }
            }
        );
    }
}

File: MSDKRemote/app/src/main/java/com/msdkremote/MyApplication.kt
package com.msdkremote

import android.app.Application
import android.content.Context
import android.util.Log
import dji.v5.common.error.IDJIError
import dji.v5.common.register.DJISDKInitEvent
import dji.v5.manager.SDKManager
import dji.v5.manager.interfaces.SDKManagerCallback

class MyApplication : Application() {

    private val TAG = this::class.simpleName

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        // 在调用install前，请勿调用任何MSDK相关
        com.secneo.sdk.Helper.install(this)
    }

    override fun onCreate() {
        super.onCreate()
        // 初始化MSDK，建议初始化逻辑放在Application中，当然也可以根据自己的需要放在任意地方。
        SDKManager.getInstance().init(this,object:SDKManagerCallback{
            override fun onInitProcess(event: DJISDKInitEvent?, totalProcess: Int) {
                Log.i(TAG, "onInitProcess: ")
                if (event == DJISDKInitEvent.INITIALIZE_COMPLETE) {
                    SDKManager.getInstance().registerApp()
                }
            }
            override fun onRegisterSuccess() {
                Log.i(TAG, "onRegisterSuccess: ")
            }
            override fun onRegisterFailure(error: IDJIError?) {
                Log.i(TAG, "onRegisterFailure: ")
            }
            override fun onProductConnect(productId: Int) {
                Log.i(TAG, "onProductConnect: ")
            }
            override fun onProductDisconnect(productId: Int) {
                Log.i(TAG, "onProductDisconnect: ")
            }
            override fun onProductChanged(productId: Int)
            {
                Log.i(TAG, "onProductChanged: ")
            }
            override fun onDatabaseDownloadProgress(current: Long, total: Long) {
                Log.i(TAG, "onDatabaseDownloadProgress: ${current/total}")
            }
        })
    }
}
